smurfs/src/components/App.js

```diff
@@ -1,22 +1,92 @@
 import React, { Component } from 'react';
 import './App.css';
-/*
- to wire this component up you're going to need a few things.
- I'll let you do this part on your own. 
- Just remember, `how do I `connect` my components to redux?`
- `How do I ensure that my component links the state to props?`
- */
+import { connect } from 'react-redux'
+import { getSmurfs, addSmurfs } from '../actions'
+import SmurfList from './SmurfList'
+
 class App extends Component {
+  state = {
+    smurfs: {
+      name: '',
+      age: '',
+      height: ''
+    }
+  }
+
+  componentDidMount() {
+    this.props.getSmurfs()
+  }
+
+  handleChanges = e => {
+    this.setState({
+      smurfs: {
+        ...this.state.smurfs,
+        [e.target.name]: e.target.value
+      }
+    })
+  }
+
+   addSmurf = e => {
+        e.preventDefault()
+        this.props.addSmurfs(this.state.smurfs)
+        this.setState({
+            smurfs: {
+                name: '',
+                age: '',
+                height: ''
+            },
+        })
+    }
+
   render() {
     return (
+      
       <div className="App">
-        <h1>SMURFS! 2.0 W/ Redux</h1>
-        <div>Welcome to your Redux version of Smurfs!</div>
-        <div>Start inside of your `src/index.js` file!</div>
-        <div>Have fun!</div>
+       <SmurfList smurfs={this.props.smurfs}/>
+        <form onSubmit={this.handleSubmit}>
```

Comment:
This form should be housed in it's own component file to keep development modular. `App.js` should have very little if no logic at all.